src/jq_test.c

```diff
@@ -490,4 +490,98 @@ static void jv_test() {
     //jv_dump(jv_copy(o2), 0); printf("\n");
     jv_free(o2);
   }
+
+  {
+	assert(jv_equal(
+			jv_paths(
+				jv_parse("{"
+					"\"key\":["
+						"{\"this\":{\"some\":true}},"
+						"false,"
+						"null,"
+						"10,"
+						"[true]"
+					"]"
+				"}")
+			),
+			jv_parse("["
+				"[\"key\"],"
+				"[\"key\", 0],"
+				"[\"key\", 0, \"this\"],"
+				"[\"key\", 0, \"this\", \"some\"],"
+				"[\"key\", 1],"
+				"[\"key\", 2],"
+				"[\"key\", 3],"
+				"[\"key\", 4],"
+				"[\"key\", 4, 0]"
+			"]")
+		));
+
+  }
+
+  {
+	assert(jv_equal(
+			jv_addpath(
```

Comment:
Ok, I think `jv_addpath()` is the same as `jv_setpath()`, so let's drop it.

Reply:
no it's not 

jv_setpath() replaces the entire path with the value you specify
jv_addpath() overwrites in the path you specify the keys specified in the replace value keeping keys that are not mentioned in the replace value intact